react/src/base/inputs/components/SprkInputElement/SprkInputElement.js

```diff
@@ -67,7 +67,7 @@ class SprkInputElement extends Component {
           'sprk-b-InputContainer__input--has-icon-right': iconRight,
         })}
         type={type}
-        htmlFor={id}
+        id={id}
```

Comment:
Instead of replacing the `htmlFor` can we add the `id={id}` while maintaining support for `htmlFor` in case someone is using it? It would mean both would render with the `id` but I think removing it and changing the API qualifies as a breaking change.

Reply:
htmlFor is only for "for" and should be put on labels. It wasn't working or affecting anything, and this is correcting it because inputs should have IDs, not `for`

```
<label htmlFor="username">Username</label>
<input name="username" id="username" type="text"/>
```

Reply:
Ok. Looks like the propType description for id needs to be updated. References htmlFor.

Reply:
I'm seeing the user isn't able to set id on this in general. Adding that functionality now

Reply:
changed!